### PR TITLE
Update Flake8 config and workflow.

### PR DESCRIPTION
**Pull request by HexDecimal**
This sets which error codes to select and ignore.  The previous config was using parts of my local configuration which was undesirable.

The GitHub workflow now uses a simpler problem matcher action and was reformatted.

Any new Flake8 errors were fixed.

### CODE REVIEW

File: delocate/delocating.py [hunk 1]
```diff
@@ -348,14 +348,12 @@ def _copy_required(
 
 def _dylibs_only(filename):
     # type: (Text) -> bool
-    return (filename.endswith('.so') or
-            filename.endswith('.dylib'))
+    return (filename.endswith('.so') or filename.endswith('.dylib'))
```
Comment (matthew-brett) [inline delocate/delocating.py:351]: I was surprised by this needed change; which rule is this?

I think we don't need the outer parentheses any more.

**Review by matthew-brett — COMMENTED** [review]
Just one comment / question.  Looks good.  Feel free to merge with or without response to comment.

File: delocate/delocating.py (hunk 1 above)
Comment (HexDecimal) [reply delocate/delocating.py:351]: The `W503,W504` codes disallow line-breaks before/after operators.  I fixed this section before deciding to ignore the rule.

**Comment by matthew-brett** [comment]
Please do merge when the tests are done ...

**Pull request merged**